osbs/core.py

```diff
@@ -44,13 +45,19 @@
 
 def check_response(response):
     if response.status_code not in (httplib.OK, httplib.CREATED):
-        if hasattr(response, 'content'):
-            content = response.content
+        if hasattr(response, 'text'):
+            text = response.text
         else:
-            content = ''.join(response.iter_lines())
+            encoding = response.encoding or 'utf-8'
```

Comment:
We could use `chardet.detect` when response.encoding is `None` -- or we could just assume UTF-8. Anyone care?